runtime/compiler/env/J9SharedCache.cpp

```diff
@@ -401,25 +406,67 @@ TR_J9SharedCache::matchRAMclassFromROMclass(J9ROMClass * clazz, TR::Compilation
 void *
 TR_J9SharedCache::pointerFromOffsetInSharedCache(void *offset)
```

Comment:
I think it would be very help to have either a doc or a comment describing how the caches are being linearized, and what the "offset" means in this multi-SCC world.

Reply:
Agreed. I'll add a doxygen comment somewhere appropriate.

Reply:
Done, added the following class doc for `TR_J9SharedCache`:

```cpp
/**
 * \brief An interface to the VM's shared class cache.
 *
 * This class provides an interface to the VM's shared class cache as represented by the the descriptors in J9SharedClassConfig::cacheDescriptorList.
 * The cache descriptor list is a circular linked list. It is doubly linked when J9VM_OPT_MULTI_LAYER_SHARED_CLASS_CACHE is defined
 * or singly linked otherwise.
 *
 * If J9VM_OPT_MULTI_LAYER_SHARED_CLASS_CACHE is not defined, the list consists of a single element who's next pointer refers back to itself.
 * Offsets into the shared cache represent the distance from the start of the cache. Converting between pointers and offsets can be done with
 * simple pointer arithmetic.
 *
 * If J9VM_OPT_MULTI_LAYER_SHARED_CLASS_CACHE is defined, the head of the list represents the top-most (and therefore writable)
 * layer of the shared class cache, the next element represents the (N-1)th layer, and so on. The element previous to the head represents the base layer.
 * The list of cache layers can be thought of as a single logical cache starting at layer 0 and ending at layer N. Offsets into the shared cache
 * represent the distance from the start of the cache. Converting between pointers and offsets requires traversing the list starting at the base layer.
 */
```

Also added method docs for the three public methods I changed.